homeassistant/components/template/binary_sensor.py

```diff
@@ -306,3 +327,84 @@ def is_on(self):
     def device_class(self):
         """Return the sensor class of the binary sensor."""
         return self._device_class
+
+
+class TriggerBinarySensorEntity(TriggerEntity, BinarySensorEntity):
+    """Sensor entity based on trigger data."""
+
+    domain = BINARY_SENSOR_DOMAIN
+    extra_template_keys = (CONF_STATE,)
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        coordinator: TriggerUpdateCoordinator,
+        config: dict,
+    ):
+        """Initialize the entity."""
+        super().__init__(hass, coordinator, config)
+
+        if isinstance(config.get(CONF_DELAY_ON), template.Template):
+            self._to_render.append(CONF_DELAY_ON)
+            self._parse_result.add(CONF_DELAY_ON)
+
+        if isinstance(config.get(CONF_DELAY_OFF), template.Template):
+            self._to_render.append(CONF_DELAY_OFF)
+            self._parse_result.add(CONF_DELAY_OFF)
+
+        self._delay_cancel = None
+        self._state = False
+
+    @property
+    def is_on(self) -> bool:
+        """Return state of the sensor."""
+        return self._state
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Handle update of the data."""
+        self._process_data()
+
+        if self._delay_cancel:
+            self._delay_cancel()
+            self._delay_cancel = None
+
+        if not self.available:
+            return
+
+        raw = self._rendered.get(CONF_STATE)
+        state = template.result_as_boolean(raw)
+
+        if state == self._state:
+            return
+
+        key = CONF_DELAY_ON if state else CONF_DELAY_OFF
+        delay = self._rendered.get(key) or self._config.get(key)
+
+        # state without delay. None means rendering failed.
+        if state is None or delay is None:
+            self._state = state
+            self.async_write_ha_state()
+            return
+
+        if not isinstance(delay, timedelta):
+            print(type(delay), delay)
+            try:
+                delay = cv.positive_time_period(delay)
+            except (template.TemplateError, vol.Invalid) as err:
```

Comment:
When do we get a template error here? Does applying the validator cause a template render?

Reply:
ah snap, that's old. And no longer possible.

Reply:
```suggestion
            except vol.Invalid as err:
```